lib/src/main/java/io/ably/lib/realtime/ChannelBase.java

```diff
@@ -131,6 +132,11 @@ private void setState(ChannelState newState, ErrorInfo reason, boolean resumed,
             this.retryCount = 0;
         }
 
+        // RTP5a1
+        if (newState == ChannelState.detached || newState == ChannelState.suspended || newState == ChannelState.failed) {
+            properties.channelSerial = null;
```

Comment:
Maybe I miss understood but [(RTN15c7)](https://sdk.ably.com/builds/ably/specification/main/features/#RTN15c7) states: "The internal msgSerial counter should be reset so that the first message published to Ably will contain a msgSerial of 0". Doesn't this mean reseted `channelSerial` is `0L` and not `null`?

Reply:
Actually fixed as a part of https://github.com/ably/ably-java/pull/980/commits/b42ff87ad98c55eeefb13d6f372e843fd5511cf2

Reply:
channelSerial is a string unline msgserial which is int